pkg/virt-api/rest/subresource.go

```diff
@@ -79,23 +80,29 @@ const (
 	volumeMigrationManualRecoveryRequiredErr = "VM recovery required: Volume migration failed, leaving some volumes pointing to non-consistent targets; manual intervention is needed to reassign them to their original volumes."
 )
 
+type instancetypeVMExpander interface {
+	Expand(vm *v1.VirtualMachine) (*v1.VirtualMachine, error)
+}
+
 type SubresourceAPIApp struct {
 	virtCli                 kubecli.KubevirtClient
 	consoleServerPort       int
 	profilerComponentPort   int
 	handlerTLSConfiguration *tls.Config
 	credentialsLock         *sync.Mutex
 	clusterConfig           *virtconfig.ClusterConfig
-	instancetypeMethods     instancetype.Methods
+	instancetypeExpander    instancetypeVMExpander
 	handlerHttpClient       *http.Client
 }
 
 func NewSubresourceAPIApp(virtCli kubecli.KubevirtClient, consoleServerPort int, tlsConfiguration *tls.Config, clusterConfig *virtconfig.ClusterConfig) *SubresourceAPIApp {
 	// When this method is called from tools/openapispec.go when running 'make generate',
 	// the virtCli is nil, and accessing GeneratedKubeVirtClient() would cause nil dereference.
-	var instancetypeMethods instancetype.Methods
+	var instancetypeExpander instancetypeVMExpander
 	if virtCli != nil {
-		instancetypeMethods = &instancetype.InstancetypeMethods{Clientset: virtCli}
+		instancetypeFinder := find.NewSpecFinder(nil, nil, nil, virtCli)
+		preferenceFinder := preferenceFind.NewSpecFinder(nil, nil, nil, virtCli)
+		instancetypeExpander = expand.New(clusterConfig, instancetypeFinder, preferenceFinder)
```

Comment:
Unrelated to the overall PR but thoughts on just passing the args to build the find implementations directly to `New` here? When extracting things I think it made sense to have fewer args to `New` but now that I'm seeing it being used I think it would be nicer to just pass `expand.New(clusterConfig, nil, nil, nil, nil, virtctl)` here etc.